bundles/org.eclipse.osgi.tests/src/org/eclipse/osgi/tests/bundles/ClassLoadingBundleTests.java

```diff
@@ -1173,13 +1165,13 @@ public void testBootGetResources01() throws Exception {
 			return; // cannot really test this if this property is set
 		// make sure there is only one manifest found
 		Bundle test = installer.installBundle("test"); //$NON-NLS-1$
-		Enumeration manifests = test.getResources("META-INF/MANIFEST.MF"); //$NON-NLS-1$
+		Enumeration<URL> manifests = test.getResources("META-INF/MANIFEST.MF"); //$NON-NLS-1$
 		assertNotNull("manifests", manifests); //$NON-NLS-1$
-		ArrayList manifestURLs = new ArrayList();
+		ArrayList<URL> manifestURLs = new ArrayList<>();
 		while (manifests.hasMoreElements())
 			manifestURLs.add(manifests.nextElement());
```

Comment:
```suggestion
		List<URL> manifestURLs = Collections.list(manifests);
```
This could be applied multiple times.

Reply:
Thanks, applied in all the places I manage to find. It all started with some annoying warning in my workspace :).

Reply:
As soon as you started to clean something up it is hard to stop :D

Reply:
> it is hard to stop

No worries, there are enough warnings for everyone ;-)

Reply:
By the way, as you both seem hunting for less code lines as well:

> Every program can be reduced by at least one line and every program contains at least one error. Therefore every program can be reduced to one line, which is wrong!

[Dan Garcia](https://news.ycombinator.com/item?id=2040602)

Reply:
> > it is hard to stop
> 
> No worries, there are enough warnings for everyone ;-)

I was getting worried that others were having all the fun!

![image](https://user-images.githubusercontent.com/208716/175522484-512a888c-fd9f-441d-ae80-3e4603597ad8.png)

Reply:
> I was getting worried that others were having all the fun!

I think there was a reason we only show 100 Warnings per default at once!